### DISCUSSION TITLE
How to port to a new NUCLEO board -- NUCLEO_H753ZI

### DISCUSSION THREAD

**Discussion by ramprao** [micropython/micropython#15710] [STM32 / Pyboard] [answered]
How does one go about porting to an unsupported NUCLEO? I'm looking for a NUCLEO with a 16 bit ADC. Th Nucleo_H753ZI is one of the models that supports it. There is the H743ZI/ZI2 but those have been obsoleted by STM. 
Thanks. 

Some useful info:
https://www.st.com/en/microcontrollers-microprocessors/stm32h743-753.html?querycriteria=productId=LN2033

The biggest difference seems to be 

> Security
> STM32H753 MCUs include the following additional security features:
> 
> Crypto/hash hardware acceleration
> Secure Firmware Install (SFI) embedded security services to authenticate and protect your software Ips while performing initial programming
> Secure Boot Secure Firmware Update (SBSFU)
>

**Comment by AndreoBotelho** [comment, marked as answer]
753 is the same 743 with adition of HW cripto, if the NUCLEO board is the same, the firmware must work without change

**Reply by ramprao** [reply]
Thanks @AndreoBotelho -- I have ordered one and will test when it comes. 

In the interim, I have copied over all of the files from the NUCLEO_H743ZI into a  NUCLEO_H753ZI directory. Also made some other changes to some other files (adc.c, mboot/main.c, and stm32.mk). After also copying over the stm32h743.ld and stm32h743_af.csv to ...753..., was able to get it to build. Looking at the ld files from the STM32CubeIDE, the .ld files looked to be the same. 

Will test the NUCLEO753 board with both firmwares when my boardgets delivered.

**Comment by ramprao** [comment]
@AndreoBotelho  -- Your suggestion of using the 743ZI firmware worked. Also my build after just copying over the NUCLEO_H743ZI files also worked. Thank you! I will close this.

**Comment by ramprao** [comment]
I'm going to reopen this. 

The firmware works but the Pins are all messed up. Any ideas on where to look to fix it. Thanks.

**Reply by ramprao** [reply]
It's been a couple of years since I last used to use Micropython. Looks like the Pin definition formalism seems to have changed at least for the newer boards as seen in pins.csv. One cannot use the older definitions such as Pin('A6') etc but must use the equivalent Pin('D12') seems to work.

**Reply by AndreoBotelho** [reply]
Pins definitions are on pins.csv, you can edit and rebuild the firmware

**Reply by ramprao** [reply]
Thanks. I did end up rebuilding it.
